server/sonar-web/src/main/js/apps/projectActivity/components/forms/AddGraphMetric.js

```diff
@@ -62,66 +58,33 @@ export default class AddGraphMetric extends React.PureComponent {
   };
 
   componentDidMount() {
-    if (this.card !== null) {
-      window.addEventListener('keydown', this.handleKey, false);
-      window.addEventListener('click', this.handleOutsideClick, false);
-    }
     this.setState({
       metrics: this.filterMetricsElements(this.props),
       selectedMetrics: this.getSelectedMetricsElements(this.props.metrics, null)
     });
   }
 
   componentWillReceiveProps(nextProps /*: Props */) {
-    if (nextProps.metrics.length > this.props.metrics.length) {
-      this.setState({
-        metrics: this.filterMetricsElements(nextProps),
-        selectedMetrics: this.getSelectedMetricsElements(nextProps.metrics, null)
-      });
-    }
-    if (nextProps.selectedMetrics.length < this.props.selectedMetrics.length) {
-      this.setState({
-        selectedMetrics: this.getSelectedMetricsElements(
-          nextProps.metrics,
-          nextProps.selectedMetrics
-        )
-      });
-    }
-  }
-
-  componentWillUnmount() {
-    window.removeEventListener('keydown', this.handleKey);
-    window.removeEventListener('click', this.handleOutsideClick);
+    this.setState({
+      metrics: this.filterMetricsElements(nextProps),
+      selectedMetrics: this.getSelectedMetricsElements(nextProps.metrics, nextProps.selectedMetrics)
+    });
   }
 
   getPopupPos = (containerPos /*: ClientRect*/) => ({
     top: containerPos.height,
     right: containerPos.width - 240
   });
 
-  handleKey = (evt /*: KeyboardEvent*/) => {
-    // Escape key
-    if (evt.keyCode === 27) {
-      this.setState({ open: false });
-    }
-  };
-
-  handleOutsideClick = (evt /*: Event*/) => {
-    // $FlowFixMe
-    if (!this.card || !this.card.contains(evt.target)) {
-      this.setState({ open: false });
-    }
-  };
-
-  filterMetricsElements = (nextProps /*: Props */) => {
-    const { metricsTypeFilter, metrics } = nextProps;
+  filterMetricsElements = (props /*: Props */) => {
+    const { metricsTypeFilter, metrics, selectedMetrics } = props;
```

Comment:
just a small tip, you can write : 
```ts
filterMetricsElements = ({metricsTypeFilter, metrics, selectedMetrics} /*: Props */) => {...
```
And remove the `const { metrics....} = props;`